ygnmi/unmarshal.go

```diff
@@ -163,6 +180,70 @@ func unmarshalAndExtract[T any](data []*DataPoint, q AnyQuery[T], goStruct ygot.
 	return ret, nil
 }
 
+func unmarshalDynamic(data []*DataPoint, val any) error {
+	switch {
+	case len(data) > 2:
+		return fmt.Errorf("got multiple datapoints for dynamic node")
+	case len(data) == 2 && !data[1].Sync:
+		return fmt.Errorf("got multiple datapoints for dynamic node")
+	}
+	rVal := reflect.ValueOf(val).Elem()
+	valType := reflect.TypeOf(val).Elem()
+	kind := valType.Kind()
+
+	switch dataVal := data[0].Value.Value.(type) {
+	case *gpb.TypedValue_StringVal:
+		if kind != reflect.String {
```

Comment:
Also check CanSet() to avoid panic? https://pkg.go.dev/reflect#Value.SetString

same for below